Objective-C/CBLCoreMLPredictiveModel.m

```diff
@@ -0,0 +1,538 @@
+    for (NSString* key in spec) {
+        MLFeatureDescription* desc = spec[key];
+        id value = [dictionary valueForKey: key];
+        if (!value) {
+            if (desc.optional)
+                continue;
+            CBLWarn(Query, @"The value of the non-optional input key (%@) is missing.", key);
+            return nil;
+        }
+        
+        BOOL isSequenceValue = NO;
+        MLFeatureType sequenceType = MLFeatureTypeInvalid;
+        if (@available(macOS 10.14, iOS 12.0, *)) {
+            isSequenceValue = (desc.type == MLFeatureTypeSequence);
+            sequenceType = desc.sequenceConstraint.valueDescription.type;
+        }
+        
+        MLFeatureValue* featureValue;
+        if (isSequenceValue) {
+            if (@available(macOS 10.14, iOS 12.0, *)) {
+                featureValue = [self.class sequenceFeatureValueFromValue: value type: sequenceType];
+            }
+        } else if (desc.type == MLFeatureTypeMultiArray) {
+            MLMultiArrayConstraint* constraint = desc.multiArrayConstraint;
+            featureValue = [self.class multiArrayFeatureValueFromValue: value
+                                                                 shape: constraint.shape
+                                                                  type: constraint.dataType] ;
+        } else {
+            featureValue = [self.class featureValueFromValue: value type: desc.type];
+        }
+        
+        if (!featureValue) {
+            CBLWarn(Query, @"Cannot convert the value of %@ key.", key);
+            return nil;
+        }
+        [dict setObject: featureValue forKey: key];
+    }
+    
+    NSError* error;
+    MLDictionaryFeatureProvider* provider =
+        [[MLDictionaryFeatureProvider alloc] initWithDictionary: dict error: &error];
+    if (!provider) {
+        CBLWarn(Query, @"Error when creating a dictionary feature provider: %@", error);
+    }
+    return provider;
+}
+
+
++ (MLFeatureValue*) featureValueFromValue: (id)value type: (MLFeatureType)type {
+    switch (type) {
+        case MLFeatureTypeInt64: {
+            NSNumber* num = $castIf(NSNumber, value);
+            return num ? [MLFeatureValue featureValueWithInt64: [num longLongValue]] : nil;
+        }
+        case MLFeatureTypeDouble: {
+            NSNumber* num = $castIf(NSNumber, value);
+            return num ? [MLFeatureValue featureValueWithDouble: [num doubleValue]] : nil;
+        }
+        case MLFeatureTypeString: {
+            NSString* str = $castIf(NSString, value);
+            return str ? [MLFeatureValue featureValueWithString: str] : nil;
+        }
+        case MLFeatureTypeDictionary: {
+            CBLDictionary* dict = $castIf(CBLDictionary, value);
+            if (!dict)
+                return nil;
+            NSError* error;
+            MLFeatureValue* dictValue = [MLFeatureValue featureValueWithDictionary:
+                                         [dict toDictionary] error: &error];
```

Comment:
styling something like this right?
<img width="751" alt="screen shot 2019-02-21 at 1 13 47 pm" src="https://user-images.githubusercontent.com/10448770/53202123-90344c80-35da-11e9-848b-24736d4558e8.png">

Reply:
OK, I will fix that.